### PR TITLE
Add even more miscellaneous NPCs.

### PR DESCRIPTION
**Pull request by MNG-cataclysm**
#### Summary
Content "Add even more miscellaneous NPCs"

#### Purpose of change
Add more assorted NPCs, to help make the world feel more dynamic.

#### Describe the solution
Add a survivor inside a gun store, a veterinarian caring for some animals, and someone held out in a private park.

#### Describe alternatives you've considered
Not adding these.

#### Testing
I spawned all locations and chatted with all NPCs. Everything works as intended.

#### Additional context
None for the moment.

### CODE REVIEW

File: data/json/npcs/other/TALK_ANIMAL_SHELTER_SURVIVOR.json [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+      },
+      {
+        "text": "Wanna buy something, maybe sell off what you don't need?",
+        "topic": "TALK_NPC_ANIMAL_SHELTER_SURVIVOR_INTRO",
+        "effect": "start_trade"
+      },
+      {
+        "text": "Why don't you come with me?",
+        "trial": { "type": "PERSUADE", "difficulty": -15, "mod": [ [ "value", 2 ] ] },
+        "success": { "topic": "TALK_AGREE_FOLLOW", "effect": "follow", "opinion": { "trust": 1, "value": 1 } },
```
Comment (Maleclypse) [inline data/json/npcs/other/TALK_ANIMAL_SHELTER_SURVIVOR.json:82]: If they've worked this hard to stay and keep the animals safe so far what's going to convince them to go along with you and potentially leave the animals behind?

Comment (MNG-cataclysm) [reply data/json/npcs/other/TALK_ANIMAL_SHELTER_SURVIVOR.json]: Nothing, you can't recruit them anymore.

File: data/json/mapgen/animalshelter.json [hunk 2]
```diff
@@ -139,6 +139,39 @@
       { "monster": "mon_dog_samoyed_pup", "weight": 5, "cost_multiplier": 0 }
     ]
   },
+  {
+    "name": "GROUP_ANIMALSHELTER_DOGS_NOZOM",
```
Comment (Night-Pryanik) [inline data/json/mapgen/animalshelter.json:143]: Nitpick, but we already have `GROUP_PET_DOGS` monster group, which has exactly the same monsters and weights as in this one.

**Comment by MNG-cataclysm** [comment]
@Maleclypse Could you remove the string freeze label, please?

**Comment by LovamKicsiGazsii** [comment]
Could you please add a flagpole to the gunstore like I did in #62787 ?

**Pull request merged**